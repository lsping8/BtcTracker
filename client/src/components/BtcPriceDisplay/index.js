import React, { Component } from 'react';

class BtcPriceDisplay extends Component {
  render() {
    const { currentPrice,currency } = this.props
    return (
        <div className="box" style={{padding: 20}}>
          <span className="currentPrice">{currency} {currentPrice}</span>
          <div className="poweredBox">Powered by CoinDesk</div>
        </div>
    );
  }
}

export default BtcPriceDisplay;
