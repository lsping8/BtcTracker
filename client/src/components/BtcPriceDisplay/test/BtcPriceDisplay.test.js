import React from 'react'
import ReactDOM from 'react-dom'
import { shallow,configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import BtcPriceDisplay from '../index'

configure({ adapter: new Adapter() });

const props = { currency: 'USD', currentPrice: 0 }
const btcPriceDisplay = shallow(<BtcPriceDisplay {...props}/>)

describe('BtcPriceDisplay', () => {
  it('renders without crashing', () => {
    expect(btcPriceDisplay).toMatchSnapshot()
  })

  it('text powered by', () => {
    expect(btcPriceDisplay.find('.poweredBox').text()).toEqual('Powered by CoinDesk')
  })

  it('initial price', () => {
    expect(btcPriceDisplay.find('.currentPrice').text()).toEqual('USD 0')
  })
})
