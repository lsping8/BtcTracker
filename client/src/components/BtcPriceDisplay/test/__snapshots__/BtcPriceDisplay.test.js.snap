// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtcPriceDisplay renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BtcPriceDisplay
    currency="USD"
    currentPrice={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="currentPrice"
>
          USD
           
          0
</span>,
        <div
          className="poweredBox"
>
          Powered by CoinDesk
</div>,
      ],
      "className": "box",
      "style": Object {
        "padding": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "USD",
            " ",
            0,
          ],
          "className": "currentPrice",
        },
        "ref": null,
        "rendered": Array [
          "USD",
          " ",
          0,
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Powered by CoinDesk",
          "className": "poweredBox",
        },
        "ref": null,
        "rendered": "Powered by CoinDesk",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="currentPrice"
>
            USD
             
            0
</span>,
          <div
            className="poweredBox"
>
            Powered by CoinDesk
</div>,
        ],
        "className": "box",
        "style": Object {
          "padding": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "USD",
              " ",
              0,
            ],
            "className": "currentPrice",
          },
          "ref": null,
          "rendered": Array [
            "USD",
            " ",
            0,
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Powered by CoinDesk",
            "className": "poweredBox",
          },
          "ref": null,
          "rendered": "Powered by CoinDesk",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
